apps/block_scout_web/lib/block_scout_web/controllers/block_epoch_transaction_controller.ex

```diff
@@ -0,0 +1,208 @@
+defmodule BlockScoutWeb.BlockEpochTransactionController do
+  use BlockScoutWeb, :controller
+
+  import BlockScoutWeb.Chain,
+    only: [paging_options: 1, next_page_params: 3, split_list_by_page: 1]
+
+  import Explorer.Chain,
+    only: [hash_to_block: 2, number_to_block: 2, string_to_address_hash: 1, string_to_block_hash: 1, hash_to_address: 1]
+
+  alias BlockScoutWeb.{Controller, EpochTransactionView}
+  alias Explorer.Celo.AccountReader
+  alias Explorer.Chain
+  alias Explorer.Chain.{CeloElectionRewards, CeloEpochRewards, Wei}
+  alias Phoenix.View
+
+  def index(conn, %{"block_hash_or_number" => formatted_block_hash_or_number, "type" => "JSON"} = params) do
+    case param_block_hash_or_number_to_block(formatted_block_hash_or_number,
+           necessity_by_association: %{
+             [miner: :names] => :required,
+             [celo_delegator: :celo_account] => :optional
+           }
+         ) do
+      {:ok, block} ->
+        paging_options_keyword = paging_options(params)
+        %Explorer.PagingOptions{page_size: page_size} = Keyword.get(paging_options_keyword, :paging_options)
+
+        epoch_transactions_plus_one =
+          CeloElectionRewards.get_paginated_rewards_for_block(block.number, Map.put(params, "page_size", page_size))
+
+        {epoch_transactions, next_page} = split_list_by_page(epoch_transactions_plus_one)
+
+        next_page_path =
+          case next_page_params(next_page, epoch_transactions, params) do
+            nil ->
+              nil
+
+            next_page_params ->
+              block_epoch_transaction_path(conn, :index, block, Map.delete(next_page_params, "type"))
+          end
+
+        carbon_fund_address_string =
+          case AccountReader.get_carbon_offsetting_partner(block.number) do
+            {:ok, address_string} -> address_string
+            :error -> "0x0ba9f5B3CdD349aB65a8DacDA9A38Bc525C2e6D6"
```

Comment:
https://explorer.celo.org/address/0x07F007d389883622Ef8D4d347b3f78007f28d8b7/read-proxy you can see the address here if you scroll down to "2. carbonOffsettingPartner". I'm getting the address from the archive node and if that fails, use the hard-coded string as fallback. This isn't ideal, because if the address changes and the archive node doesn't respond, we will show the wrong address. I'm not sure how to best handle the error here.

Reply:
Can we assign this variable a meaningful name then? It's not clear from the code that this address represents the mainnet carbon offsetting partner.

Reply:
I went a different route and added an error message. Would make no sense to display the wrong address.